kotest-runner/kotest-runner-junit5/src/jvmMain/kotlin/io/kotest/runner/junit/platform/gradle/GradleClassMethodRegexTestFilter.kt

```diff
@@ -13,70 +14,85 @@ class GradleClassMethodRegexTestFilter(private val patterns: List<String>) : Tes
       logger.log { Pair(descriptor.toString(), "Testing against $patterns") }
       return when {
          patterns.isEmpty() -> TestFilterResult.Include
-         patterns.any { match(it, descriptor) } -> TestFilterResult.Include
```

Comment:
This was wrongly combining the filters. For instance if we have two in the list `*Test` defined in build.kts and `CacheTest` in command line, you'd want to omit all other tests except for `CacheTest`. This flip from `any` to `all` fixes it.